src/Collection/CollectionTrait.php

```diff
@@ -95,6 +96,18 @@ public function reject(callable $callback): CollectionInterface
         return new FilterIterator($this->unwrap(), fn ($key, $value, $items) => !$callback($key, $value, $items));
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function unique(?callable $callback = null): CollectionInterface
+    {
```

Comment:
Does this default have any practical use case? If not the argument shouldn't be nullable.

Reply:
I'm wondering if none of filter, reject and unique should have a default.

Reply:
Filtering our falsy values is a pretty common use case, so having filter() do that by default is useful.

Reply:
Ok, I can remove the default for this if we don't think they all need one.